packages/ag-charts-enterprise/src/features/annotations/parallel-channel/parallelChannelScene.ts

```diff
@@ -63,36 +63,60 @@ export class ParallelChannelScene extends ChannelScene<ParallelChannelProperties
         }
     }
 
-    override dragHandle(datum: ParallelChannelProperties, target: Coords, context: AnnotationContext) {
+    override dragHandle(
+        datum: ParallelChannelProperties,
+        target: Coords,
+        context: AnnotationContext,
+        shiftKey: boolean
+    ) {
         const { activeHandle, handles } = this;
         if (activeHandle == null) return;
 
         const { offset } = handles[activeHandle].drag(target);
         handles[activeHandle].toggleDragging(true);
 
         const prev = datum.toJson();
-        let moves: Array<ChannelHandle> = [];
+        let moves: Array<[ChannelHandle, ChannelHandle | undefined]> = [];
```

Comment:
Perhaps have a new variable rather than making this a nested array, it's a bit messy.

```
let moves: Array<ChannelHandle> = [];
let origins: Array<ChannelHandle> = [];

// ...

switch (activeHandle) {
    case 'topLeft':
    case 'bottomLeft':
        moves = ['topLeft', 'bottomLeft'];
        origins = ['topRight', 'bottomRight'];
        break;

// ...
```